File: Data Structures & Algorithms/longest-consecutive-sequence/submission-7.cpp
#include <cmath>
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        int c = 0;
        unordered_set<int> s;
        s.insert(nums.begin(), nums.end());
        for(int i: nums) {
            int l = 1;
            while (s.count(i+l) == 1)
                l++;
            c = max(c, l);
        }
        return c;
    }
};

File: Data Structures & Algorithms/trapping-rain-water/submission-7.py
class Solution:
    def trap(self, height: List[int]) -> int:
        if not height:
            return 0
        l, r, m = 0, len(height)-1, 0
        lm, rm = height[l], height[r]
        while l < r:
            if height[l] < height[r]:
                lm = max(lm, height[l])
                m+= lm- height[l]
                l+=1
            else:
                rm = max(rm, height[r])
                m+= rm- height[r]
                r-=1
        return m